src/payment/strategies/sdk/strategy.ts

```diff
@@ -0,0 +1,155 @@
+
+*/
+
+export class Strategy implements PaymentStrategy {
+    protected _paymentProcessor?: PaymentProcessor;
+
+    constructor(
+        protected _store: CheckoutStore,
+        protected _orderActionCreator: OrderActionCreator
+    ) {}
+
+    async initialize(options?: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        if (!options || !options.methodId) {
+            // TODO: Can we make this object and attribute mandatory?
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        const { methodId } = options;
+
+        /*
+            TODO: probably too brittle to just use methodId going forward
+
+            if methodId is 'card', how do we differentiate between standard and "has own JS" implementations?
+
+            N.B. looking for a "has own JS" initializer, not finding one,
+            then falling back to a default one for that methodId
+            could produce some unpredictable (and likely broken) UX during execute
+        */
+        const initializer = initializers[methodId];
+
+        if (!initializer) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        // Suggestions for better names welcome :D
+        const paymentProcessor = await initializer({
+            options,
+            store: this._store,
+        });
+
+        if (!paymentProcessor) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        // Sore the processor for later use in execute
+        this._paymentProcessor = paymentProcessor;
+
+        return this._store.getState();
+    }
+
+    deinitialize(_options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        // Nothing to be done here?
+        return Promise.resolve(this._store.getState());
+    }
+
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+
+        if (!payment || !payment.methodId) {
+            throw new PaymentArgumentInvalidError(['payment.methodId']);
+        }
+
+        const paymentProcessor = this._paymentProcessor;
+
+        if (!paymentProcessor) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        // Submitting the order should let us restore it post redirect in finalize
+        const submitOrder = () => this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+        const getPaymentStep = () => paymentProcessor(payment);
+        const loadOrder = () => this._store.dispatch(this._orderActionCreator.loadCurrentOrder());
+
+        // TODO: move to async/await format?
+        return submitOrder()
+            // Very key differences to a typical strategy:
+            .then(getPaymentStep)
```

Comment:
I wonder if there could be multiple payments step, in which case after submitting the order, you might need some sort of recursive call that processes all the steps until it reaches the end.

Basically something like
```js
function processStep() {
  step = getPaymentStep();
  if (handler(step)) {
    return;
  } else {
    processStep();
  }

}

Reply:
Yep, we'll definitely have something like that to handle the multiple steps.

https://github.com/bigcommerce/checkout-sdk-js/pull/1108/files#diff-34830b76d352231339db788439c1039d6d412c2ee859dca4b4f77bfbf09703bcR42

The current thinking is that the 'handler' would pass itself to any function that carries out a step.
So that if/when that step's executions ends in another response from a processing endpoint, that response is passed once again to the handler.

I guess from the point of the Strategy file, this recursion is hidden within the internals of the handler.